### PR TITLE
delete stack operation for openstack

### PR DESCRIPTION
**Pull request by drmaas**
delete stack servergroup works

start adding tests, need to rebase

start adding tests, need to rebase

finished unit tests for delete server group

fixing header

fixing some leftover merge issues

### CODE REVIEW

**Comment by drmaas** [comment]
@spinnaker/reviewers

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/servergroup/DeployOpenstackAtomicOperationValidator.groovy [hunk 1]
```diff
@@ -1,30 +1,32 @@
 /*
- * Copyright 2016 Veritas Technologies LLC.
+ * Copyright 2016 Target, Inc.
  *
  * Licensed under the Apache License, Version 2.0 (the "License");
  * you may not use this file except in compliance with the License.
  * You may obtain a copy of the License at
  *
- *    http://www.apache.org/licenses/LICENSE-2.0
+ *     http://www.apache.org/licenses/LICENSE-2.0
```
Comment (ashleykasim) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/servergroup/DeployOpenstackAtomicOperationValidator.groovy]: Not sure why the spacing diffs in the header

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/servergroup/DeployOpenstackAtomicOperationValidator.groovy [hunk 2]
```diff
@@ -1,30 +1,32 @@
...
+import com.netflix.spinnaker.clouddriver.openstack.deploy.validators.OpenstackAttributeValidator
 import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperations
 import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
 import org.springframework.beans.factory.annotation.Autowired
 import org.springframework.stereotype.Component
 import org.springframework.validation.Errors
 
+//TODO this needs to be tested
 @OpenstackOperation(AtomicOperations.CREATE_SERVER_GROUP)
 @Component("deployOpenstackAtomicOperationValidator")
```
Comment (ashleykasim) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/servergroup/DeployOpenstackAtomicOperationValidator.groovy]: Use just `@Component`

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/servergroup/DestroyOpenstackAtomicOperationValidator.groovy [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+import com.netflix.spinnaker.clouddriver.openstack.deploy.description.servergroup.DestroyOpenstackAtomicOperationDescription
+import com.netflix.spinnaker.clouddriver.openstack.deploy.validators.OpenstackAttributeValidator
+import com.netflix.spinnaker.clouddriver.orchestration.AtomicOperations
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Component
+import org.springframework.validation.Errors
+
+@OpenstackOperation(AtomicOperations.DESTROY_SERVER_GROUP)
+@Component("destroyOpenstackAtomicOperationValidator")
```
Comment (ashleykasim) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/deploy/validators/servergroup/DestroyOpenstackAtomicOperationValidator.groovy]: same here, `@Component` only

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy [hunk 4]
```diff
@@ -177,13 +171,34 @@ abstract class OpenstackClientProvider {
...
+      client.heat().stacks().list()
+    }
+  }
+
+  /**
+   * Get a stack in a specific region.
+   * @param stackName
+   * @return
+   */
+  Stack getStack(String region, String stackName) {
```
Comment (ashleykasim) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy]: This duplicates the `getServerGroup(region, stackName` introduced in the clone server group PR. Please consolidate these methods

File: clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy [hunk 5]
```diff
@@ -177,13 +171,34 @@ abstract class OpenstackClientProvider {
...
+      throw new OpenstackProviderException("Unable to find stack $stackName in region $region")
+    }
+    stack
+  }
+
+  /**
+   * Delete a stack in a specific region.
+   * @param stack
+   */
+  ActionResponse destroy(String region, Stack stack) {
```
Comment (ashleykasim) [inline clouddriver-openstack/src/main/groovy/com/netflix/spinnaker/clouddriver/openstack/client/OpenstackClientProvider.groovy]: shouldn't the return type be `void`?

**Pull request closed without merging**